test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/initialgainorloss/InitialGainOrLossControllerSpec.scala

```diff
@@ -0,0 +1,342 @@
+            doc.select("#content > article > form").attr("action") shouldBe routes.InitialGainOrLossController
+              .checkYourAnswersSubmit()
+              .url
+          },
+          OK
+        )
+      }
+    }
+  }
+  "InitialGainOrLossController" when {
+    def performAction(): Future[Result] = controller.enterInitialGainOrLoss()(FakeRequest())
+
+    "initialGainOrLoss is not present in draftReturn" should {
+
+      val fillingOutReturn =
+        fillingOutReturnSample.copy(draftReturn = fillingOutReturnSample.draftReturn.copy(initialGainOrLoss = None))
+
+      "display the page with backlink to tasklist" in {
+
+        inSequence {
+          mockAuthWithNoRetrievals()
+          mockGetSession(
+            SessionData.empty.copy(
+              journeyStatus = Some(fillingOutReturn)
+            )
+          )
+        }
+        checkPageIsDisplayed(
+          performAction(),
+          messageFromMessageKey(
+            "initialGainOrLoss.title"
+          ), { doc =>
+            doc.select(".govuk-caption-xl").html()  should include("Initial gain or loss")
+            doc.select("#initialGainOrLoss").html() should include("Did you make an initial gain or loss?")
+            doc.select("#back").attr("href")        shouldBe returns.routes.TaskListController.taskList().url
+            doc.select("#content > article > form").attr("action") shouldBe routes.InitialGainOrLossController
+              .submitInitialGainOrLoss()
+              .url
+          },
+          OK
+        )
+      }
+
+    }
+    "initialGainOrLoss is present in returnDraft" should {
+      val fillingOutReturnWithGain =
+        fillingOutReturnSample.copy(draftReturn =
+          fillingOutReturnSample.draftReturn.copy(initialGainOrLoss = Some(AmountInPence(300)))
+        )
+
+      "display the page with backlink to check your answers" in {
+
+        inSequence {
+          mockAuthWithNoRetrievals()
+          mockGetSession(
+            SessionData.empty.copy(
+              journeyStatus = Some(fillingOutReturnWithGain)
+            )
+          )
+        }
+        checkPageIsDisplayed(
+          performAction(),
+          messageFromMessageKey(
+            "initialGainOrLoss.title"
+          ), { doc =>
+            doc.select(".govuk-caption-xl").html()  should include("Initial gain or loss")
+            doc.select("#initialGainOrLoss").html() should include("Did you make an initial gain or loss?")
+            doc.select("#back").attr("href")        shouldBe routes.InitialGainOrLossController.checkYourAnswers().url
+            doc.select("#content > article > form").attr("action") shouldBe routes.InitialGainOrLossController
+              .submitInitialGainOrLoss()
+              .url
+          },
+          OK
+        )
+      }
+    }
+
+    "there is an error updating the draft return" should {
+      def performAction(data: (String, String)*): Future[Result] =
```

Comment:
it's still a bit difficult to follow the spec to see what's been tested for each endpoint. Could we go for something like
```scala
"InitialGainOrLossController" when {
   "handling requests to display the initial gain or loss page" must {
    def performAction() = ...
    behave like redirectToStartBehaviour(...)
    "display the page with..." when {
       "things" in {}
    }


"handling submitted answers to the display the initial gain or loss page" must {
    def performAction() = ...
    behave like redirectToStartBehaviour(...)
    "show a form error" when {
       "things" in {}
    }

   }

"handling requests to display the check your answers page" must {
    def performAction() = ...
    behave like redirectToStartBehaviour(...)

      "redirect to places" when {
        "the section is not complete" in {}
     }

     "show the page" when {
        "the section is complete" in {}
       }

   }
   }
}
```

The specs can get quite big, so having a good structure is important to be able to navigate through them.

Reply:
Done